src/main/kotlin/com/github/eltonvs/obd/connection/ObdDeviceConnection.kt

```diff
@@ -45,32 +45,37 @@ class ObdDeviceConnection(private val inputStream: InputStream, private val outp
         return ObdRawResponse(rawData, elapsedTime)
     }
 
-    private fun sendCommand(command: ObdCommand, delayTime: Long = 0) {
-        outputStream.write("${command.rawCommand}\r".toByteArray())
-        outputStream.flush()
-        if (delayTime > 0) {
-            sleep(delayTime)
+    private suspend fun sendCommand(command: ObdCommand, delayTime: Long = 0) = runBlocking {
+        withContext(Dispatchers.IO) {
+            outputStream.write("${command.rawCommand}\r".toByteArray())
+            outputStream.flush()
+            if (delayTime > 0) {
+                sleep(delayTime)
+            }
         }
     }
 
-    private fun readRawData(): String {
+    private suspend fun readRawData(): String = runBlocking {
         var b: Byte
         var c: Char
         val res = StringBuffer()
 
-        // read until '>' arrives OR end of stream reached (-1)
-        while (inputStream.available() > 0) {
-            b = inputStream.read().toByte()
-            if (b < 0) {
-                break
-            }
-            c = b.toChar()
-            if (c == '>') {
-                break
+        withContext(Dispatchers.IO) {
+            // read until '>' arrives OR end of stream reached (-1)
+            while (inputStream.available() > 0) {
+                //println("InputStream Available: ${inputStream.available()}")
```

Comment:
Nit: Can we remove this commented-out line?

Reply:
No problem, will get it removed. Thank you for the review.